omero/annotation_scripts/Export_to_csv.py

```diff
@@ -0,0 +1,582 @@
+    # End sorting, start concatenation
+
+    res_rows = []
+    for idx in indexes:
+        curr_row = [str(obj_id_l[idx])] + [obj_name_l[idx]] + rows[idx]
+        if with_parents:
+            curr_row = [e[1] for e in obj_ancestry_l[idx]] + curr_row
+        res_rows.append(curr_row)
+    header_row.insert(0, "OBJECT_ID")
+    header_row.insert(1, "OBJECT_NAME")
+    ns_row.insert(0, "")
+    ns_row.insert(1, "")
+
+    if with_parents:
+        i = 0
+        while "" in [e[0] for e in obj_ancestry_l[i]]:
+            i += 1  # Find the row with complete parent names
+        for j in range(len(obj_ancestry_l[i])):
+            header_row.insert(j, obj_ancestry_l[i][j][0].upper())
+            ns_row.insert(j, "")
+    ns_row[0] = "NAMESPACE"
+
+    print(f"\tColumn names: {header_row}", "\n")
+
+    return ns_row, header_row, res_rows
+
+
+def attach_csv(conn, obj_, rows, separator, csv_name):
+    if not obj_.canAnnotate() and WEBCLIENT_URL == "":
+        for row in rows:
+            print(f"{separator.join(row)}")
+        return None
+
+    # create the tmp directory
+    tmp_dir = tempfile.mkdtemp(prefix='MIF_meta')
+    (fd, tmp_file) = tempfile.mkstemp(dir=tmp_dir, text=True)
+    tfile = os.fdopen(fd, 'w', encoding="utf-8")
+    for row in rows:
+        tfile.write(f"{separator.join(row)}\n")
+    tfile.close()
+
+    # link it to the object
+    file_ann = conn.createFileAnnfromLocalFile(
+        tmp_file, origFilePathAndName=csv_name,
+        ns='KeyVal_export')
+
+    if obj_.canAnnotate():
+        obj_.linkAnnotation(file_ann)
+        print(f"{file_ann} linked to {obj_}")
+
+    # remove the tmp file
+    os.remove(tmp_file)
+    os.rmdir(tmp_dir)
+
+    return file_ann.getFile()
+
+
+def run_script():
+    """
+    The main entry point of the script, as called by the client via the
+    scripting service, passing the required parameters.
+    """
+
+    # Cannot add fancy layout if we want auto fill and selct of object ID
+    source_types = [
+                    rstring("Project"), rstring("Dataset"), rstring("Image"),
+                    rstring("Screen"), rstring("Plate"), rstring("Well"),
+                    rstring("Acquisition"), rstring("Image"), rstring("Tag"),
+    ]
+
+    # Duplicate Image for UI, but not a problem for script
+    target_types = [
+                    rstring("<on current>"), rstring("Project"),
+                    rstring("- Dataset"), rstring("-- Image"),
+                    rstring("Screen"), rstring("- Plate"),
+                    rstring("-- Well"), rstring("-- Acquisition"),
+                    rstring("--- Image")
+    ]
+
+    separators = [";", ",", "TAB"]
+    # Here we define the script name and description.
+    # Good practice to put url here to give users more guidance on how to run
+    # your script.
+    client = scripts.client(
+        'Export to CSV',
+        """
+    This script exports for the selected objects their name, IDs and associated
+    key-value pairs.
```

Comment:
Could add "Tags" to this description now?

Reply:
added to export and import scripts description